File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int n = nums.length;
        
        if (n == 1) {
            return true;
        }

        boolean smaller = nums[0] < nums[n - 1];

        for (int i = 1; i < n; i++) {
            if (nums[i] < nums[i - 1]) {
                if (smaller) {
                    return false;
                }
                smaller = true;
            }
        }

        return true;
    }
}

File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        n = len(nums)
        if n == 1:
            return True
        smaller = nums[0] < nums[-1]

        for i in range(1, n):
            if nums[i] < nums[i - 1]:
                if smaller:
                    return False
                smaller = True
        return True